cf/test/test_Data.py

```diff
@@ -131,6 +131,293 @@ class DataTest(unittest.TestCase):
     #    test_only = ['test_Data_clip']
     #    test_only = ['test_Data__init__dtype_mask']
 
+    def test_Data_equals(self):
+        if self.test_only and inspect.stack()[0][3] not in self.test_only:
+            return
+
+        shape = 3, 4
+        a = np.arange(12).reshape(*shape)
+
+        d = cf.Data(a, "m")
+        self.assertTrue(d.equals(d))  # also do self-equality checks!
```

Comment:
In general , should do self equality with `d` and `d.copy()`, as the test short circuits if `self is other`: https://github.com/NCAS-CMS/cfdm/blob/master/cfdm/mixin/container.py#L213
```suggestion
        self.assertTrue(d.equals(d.copy()))  # also do self-equality checks!
```

Reply:
That's wise, very good point. Thanks, I'll get those statements all converted in that way.

Reply:
... although the "are Nan/infs equal" question may have some bearing on using `self is other`